mava/components/jax/updating/checkpointer.py

```diff
@@ -0,0 +1,96 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import List, Type
+
+from acme.jax import savers as acme_savers
+from chex import dataclass
+
+from mava.callbacks import Callback
+from mava.components.jax import Component
+from mava.components.jax.updating.parameter_server import ParameterServer
+from mava.core_jax import SystemParameterServer
+from mava.wrappers import SaveableWrapper
+
+"""Checkpointer component for Mava systems."""
+
+
+@dataclass
+class CheckpointerConfig:
+    checkpoint_minute_interval: float = 5
+
+
+class Checkpointer(Component):
+    def __init__(
+        self,
+        config: CheckpointerConfig = CheckpointerConfig(),
+    ):
+        """Component for checkpointing system variables."""
+        self.config = config
+
+    def on_parameter_server_init(self, server: SystemParameterServer) -> None:
+        """Create the system checkpointer.
+
+        Args:
+            server: SystemParameterServer.
+
+        Returns:
+            None.
+        """
+        saveable_parameters = SaveableWrapper(server.store.parameters)
+        server.store.system_checkpointer = acme_savers.Checkpointer(
+            object_to_save=saveable_parameters,  # must be type saveable
+            directory=server.store.experiment_path,
+            add_uid=False,
+            time_delta_minutes=0,
+        )
+        server.store.last_checkpoint_time = time.time()
+        server.store.checkpoint_minute_interval = self.config.checkpoint_minute_interval
+
+    def on_parameter_server_run_loop_checkpoint(
+        self, server: SystemParameterServer
+    ) -> None:
+        """Intermittently checkpoint the server parameters.
+
+        Args:
+            server: SystemParameterServer.
+
+        Returns:
+            None.
+        """
+        if (
+            time.time() - server.store.last_checkpoint_time
+            > self.config.checkpoint_minute_interval * 60 + 1
+        ):
+            server.store.system_checkpointer.save()
+            server.store.last_checkpoint_time = time.time()
+
+    @staticmethod
+    def name() -> str:
+        """Static method that returns component name."""
+        return "checkpointer"
+
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        ParameterServer required to set up server.store.parameters
+        and server.store.experiment_path.
+
+        Returns:
+            List of required component classes.
+        """
+        return [ParameterServer]
```

Comment:
Created a new component for the checkpointer that uses param server parameters. I believe that their purposes are different: the server handles maintaining the parameters and the checkpointer handles saving checkpoints

Reply:
Sounds good to me. Do you think they would every be used without one another? Would having them as one component or two components be more readable?